### ISSUE TITLE
Uncaught (in promise) TypeError: Cannot set property 'uuid' of undefined

### ISSUE THREAD

**Issue by chanakamin** [OneSignal/OneSignal-Website-SDK#258]
Hello,I started to get this error:
Uuid.ts:16 Uncaught (in promise) TypeError: Cannot set property 'uuid' of undefined
    at t (Uuid.ts:16)
    at new <anonymous> (prototype.js:391)
    at e.<anonymous> (Database.ts:223)
    at i (tslib.es6.js:91)
    at Object.next (tslib.es6.js:72)
    at s (tslib.es6.js:62)
    at <anonymous>.

and don't see anything.
{
            appId: "APP-ID",
            autoRegister: true,
            subdomainName: 'sub-name',
            persistNotification: false,
            notifyButton: {
                enable: true,
                size: 'large',
                theme: 'default',
                position: 'bottom-left',
                offset: {
                    bottom: '30px',
                    right: '30px'
                },
                colors: { // Customize the colors of the main button and dialog popup button
                    'circle.background': 'rgb(84,110,123)',
                    'circle.foreground': 'white',
                    'badge.background': 'rgb(84,110,123)',
                    'badge.foreground': 'white',
                    'badge.bordercolor': 'white',
                    'pulse.color': 'white',
                    'dialog.button.background.hovering': 'rgb(77, 101, 113)',
                    'dialog.button.background.active': 'rgb(70, 92, 103)',
                    'dialog.button.background': 'rgb(84,110,123)',
                    'dialog.button.foreground': 'white'
                },
                prenotify: true,
                showCredit: false,
                text: {
                    'tip.state.unsubscribed': 'Subscribe to notifications',
                    'tip.state.subscribed': "You're subscribed to notifications",
                    'tip.state.blocked': "You've blocked notifications",
                    'message.prenotify': 'Customize your messages',
                    'message.action.subscribed': "Thanks for subscribing!",
                    'message.action.resubscribed': "You're subscribed to notifications",
                    'message.action.unsubscribed': "You won't receive notifications again",
                    'dialog.main.title': "Customize your messages",
                    'dialog.main.button.subscribe': 'SUBSCRIBE',
                    'dialog.main.button.unsubscribe': 'UNSUBSCRIBE' ,
                    'dialog.blocked.title': 'Unblock Notifications',
                    'dialog.blocked.message': "Follow these instructions to allow notifications:"
                }
            },
            safari_web_id: `'web-id',`
            promptOptions: {
                actionMessage: "We'd like to show you notifications about your cart.",
                acceptButtonText: "ALLOW",
                cancelButtonText: "NO THANKS"
            }
        }
My config object is:

I use http site.

**Comment by jasonpang** [comment]
Hi @chanakamin,

We've modified our SDK to avoid triggering Prototype.js's bug on this error.

Prototype.js v1.7 overwrites the browser's native `Function.bind` in a way that causes valid code to crash.

When we transpile this ES7 code using TypeScript's transpiler:

`new SomeClass(await someAsyncCall())`

The resulting ES5 code uses `Function.bind` in a way that conflicts with prototype's v1.7 implementation.

Prototype v1.7.3 fixes the issue, and we would definitely recommend upgrading to the latest prototype version if possible. We've also changed the way we write that statement to avoid triggering Prototype v1.7's error.

Prototype v1.7 was released back in 2010, so it's about 7 years old!
